File: fixtures/python/active_debug_code/negative.py
# Fixture: MAINT011-active-debug-code — Python negative cases
# None of these should produce a MAINT011 finding.

import logging

logger = logging.getLogger(__name__)


def process_data(data):
    logger.info("Processing data")
    return data


def handle_error(err):
    logger.error("Error occurred: %s", err)


# print inside __main__ guard — not flagged
if __name__ == "__main__":
    print("Running in development mode")

File: fixtures/js/active_debug_code/negative.ts
// Fixture: MAINT011-active-debug-code — JS/TS negative cases
// None of these should produce a MAINT011 finding.

function handleError(err: Error): void {
    console.error("Request failed:", err); // NOT flagged — legitimate error output
    console.warn("Deprecated API used"); // NOT flagged — legitimate warning
    console.info("Server listening on :8080"); // NOT flagged — legitimate info
}

function computeSum(a: number, b: number): number {
    return a + b;
}
